subcommands/el2g/gateway.go

```diff
@@ -26,17 +28,19 @@ func init() {
 	}
 	cmd.AddCommand(configCmd)
 	configCmd.Flags().StringVarP(&pkiDir, "pki-dir", "", "", "Directory container factory PKI keys")
+	configCmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Load a root CA key on a PKCS#11 compatible HSM using this module")
+	configCmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to log into the HSM")
+	configCmd.Flags().StringVarP(&hsmTokenLabel, "hsm-token-label", "", "", "The label of the HSM token containing the root CA key")
 }
 
 func doDeviceGateway(cmd *cobra.Command, args []string) {
 	factory := viper.GetString("factory")
 
-	path := filepath.Join(pkiDir, "factory_ca.key")
-	_, err := os.Stat(path)
-	subcommands.DieNotNil(err)
-	path = filepath.Join(pkiDir, "sign_ca_csr")
-	_, err = os.Stat(path)
+	subcommands.DieNotNil(os.Chdir(pkiDir))
+	hsm, err := x509.ValidateHsmArgs(
+		hsmModule, hsmPin, hsmTokenLabel, "--hsm-module", "--hsm-pin", "--hsm-token-label")
```

Comment:
From just reading this code, it now looks as if using a hsm is mandatory. However we then validate the args and when supplied with empty strings, then we still continue to pass a `nil` value that will get ignored in `x509.InitHSM`.

Reply:
At first I wanted to do `if hsm != nil { x509.InitHsm(hsm) }`, but that increases the amount of code.
So I decided to trade off some clarity for space, that seems forgivable here.